BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/distance/BeaconDistanceCalculator.java

```diff
@@ -23,6 +23,22 @@ public static float calculateDistanceTo(Beacon beacon) {
         return calculateDistanceTo(beacon, beacon.getFilteredRssi());
     }
 
+    /**
+     * Use Pythagoras with altitude to calculate more accurate distance to the beacon, if the
+     * distance is larger than the altitude.
+     */
+    public static float calculateDistanceWithAltitudeTo(Beacon beacon, float rssi) {
+        double altitude = beacon.getLocation().getAltitude();
+        double distance = calculateDistanceTo(beacon, rssi);
```

Comment:
Why do you expect a `double` here when `calculateDistanceTo(...)` only returns `float`? You can avoid casting back to `float` later.